packages/react-portal/Spec.md

```diff
@@ -0,0 +1,158 @@
+
+Meanwhile v0 does not allow any kind of way to disable native React event bubbling through portals, to achieve a similar effect, the user must manage this themselves.
+
+#### DOM insertion
+
+Both `Layer` and `Portal` allow insertion of portals to the same part of the DOM element.
+
+v8 Layer does this through a `LayerHost` which can be rendered at any part of the React tree. The result is a HTMLDiv element with a specific CSS class and unique Id. `Layer` will attempt to find a `LayerHost` to mount either by CSS class or user provided `hostId`. The default fallback is `document.body`. Each `Layer` will render its own `Fabric` provider. The `insertFirst` property supported by `Layer` was introduced in #8065 for modeless Dialogs which was achieved through `position: static` and DOM insertion order. The same effect can be achieved through `pointer-events: none`.
+
+The v0 `PortalBoxContext` stores a single HTMLDiv that is usually a child of `document.body` where all `Portal` components are rendered by default. The v0 `Provider` is rendered for the default portal element, where style and RTL overrides could be applied for all portals within.
+
+#### Lifecycle methods
+
+v0 `Portal` only supports the following lifecycle methods:
+
+- onMount
+- onUnmount
+
+`Layer` supports the following:
+
+- onLayerMounted
+- onLayerWillUnmount
+
+The lifecycle events are very similar, with the only difference being that `onLayerWillUnmount` is called with `hostId` changes. v0 does not consider the mountNode changing as an unmount of the component itself.
+
+#### Focus Management
+
+v0 `Portal` can be configured to focus trap its contents while v8 `Layer` does not offer this and users would need to use `FocusTrapZone` for this purpose.
+
+## Sample Code
+
+`Portal` by default mounts the content to `document.body`. In the event a consumer needs to target a specific mount node for Portal content this should be configurable via prop. Both variants should still be able to access theme and fluent context if available.
+
+```
+const customElement = document.createElement('div');
+
+<App> // using FluentProvider of ThemeProvider but not PortalProvider
+  <Portal /> // attached to document.body
+  <Portal mountNode={customElement} /> // mounted on custom element
+</App>
+```
+
+`Portal` should be used as a component at any part of the React tree:
+
+```tsx
+<ContextProvider>
+  <Portal>
+    <ContextConsumer /> // should be able to access context
+  </Portal>
+</ContextProvider>
+```
+
+`Portal` should be able to access theme values as css variables:
+
+```tsx
+const useStyles = makeStyles({
+    portalContent: theme => {...}
+})
+
+
+const styles = useStyles();
+
+<ThemeProvider>
+    <Portal className={styles.portalContent>
+      Can use all theme CSS variables from the parent ThemeProvider
+    </Portal>
```

Comment:
(nit) In the current design, the className would have to go on a child element, right?
```suggestion
    <Portal>
      <div className={styles.portalContent}>
        Can use all theme CSS variables from the parent ThemeProvider
      </div>
    </Portal>
```

Reply:
Yes sorry, there's been a few changes